### PR TITLE
Add Enterprise Docs

### PR DESCRIPTION
**Pull request by rossmcdonald**

### CODE REVIEW

File: articles/enterprise/components/api-overview.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
+# The Stoplight API
+
+The __API__ component powers the Stoplight backend, connecting the UI to the datastore and other miscellaneous Stoplight services.
+
+> #### Networking Details
+>
+> The default port for the API component is TCP port __3030__. The port can be customized using the `PORT` environment variable.
+> 
+> Be sure that the Gitlab component is available __before__ starting the API service.
```
Comment (rowa97) [inline articles/enterprise/components/api-overview.md]: GitLab is camelcase

File: articles/enterprise/components/installation-overview.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The Stoplight Enterprise platform provides a fully-functional on-premise API design and documentation toolkit, taking the hassle out of your API strategy.
+
+## Deployment Options
+
+Before beginning with the installation, be sure to prepare all of the necessary systems that will be running Stoplight components. For more information on sizing, disk, and network requirements, please see the [Technical Requirements](/technical-requirements) section.
+
+### Single-server vs. Multi-server Deployments
+
+Stoplight can be deployed on one or many Linux servers (dedicated or virtualized). Single-server deployments run all of the necessary Stoplight components on a single Linux instance. This greatly simplifies the deployment process, as all components do not have to reach over the network to talk to eachother. Despite ease of installation, there are some notable shortcomings to this option:
```
Comment (rowa97) [inline articles/enterprise/components/installation-overview.md]: "...have to reach over the network to talk to one another."

File: articles/enterprise/components/api-overview.md (hunk 1 above)
Comment (rowa97) [reply articles/enterprise/components/api-overview.md]: "Make sure that the..."

File: articles/enterprise/components/api-overview.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+### Docker Installation
+
+To install the API component with Docker, run the command below:
+
+```bash
+docker pull quay.io/stoplight/api
+```
+
+> Note, if you have not already authenticated with the Stoplight container registry, you will be prompted for credentials
```
Comment (rowa97) [inline articles/enterprise/components/api-overview.md]: a bit awkwardly phrased, what needs to be authenticated? you, your machine?

File: articles/enterprise/components/api-overview.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+# SL_GITLAB_HOST is the full URL to the Stoplight Gitlab instances HTTP port.
+SL_GITLAB_HOST="http://localhost:8080"
+
+# SL_REDIS_URL is the full URL to a Redis instance.
+SL_REDIS_URL="redis://:password@exampl.com:6379"
+```
+
+> Please note that the `SIGN_SECRET` environment variable must remain static between service restarts
+
+Be sure to customize any variable above as needed.
```
Comment (rowa97) [inline articles/enterprise/components/api-overview.md]: "...any of the variables above as needed"

File: articles/enterprise/components/api.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+docker pull quay.io/stoplight/api
+```
+
+> Note, if the system you are using has not already authenticated with the
+> Stoplight container registry, you will be prompted for credentials.
+
+## Configuration
+
+To configure the Stoplight API component, you will need to provide runtime
+values and connection details to the other necessary Stoplight components. The
```
Comment (rowa97) [inline articles/enterprise/components/api.md]: "the" configuration file

File: articles/enterprise/components/api.md [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+
+Prior to installing the RPM package, you will need to:
+
+* Install NodeJS
+
+* Have the Stoplight package repository installed and configured with your user-specific credentials
+
+#### Installing NodeJS
+
+To install NodeJS, please run the following commands:
```
[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/api.md]: pretty please with sprinkles on top? 😄

[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/api.md]: Fixed

File: articles/enterprise/components/api.md (hunk 5 above)
Comment (rossmcdonald) [reply articles/enterprise/components/api.md]: Fixed

File: articles/enterprise/components/app.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+either by the configuration file or through the environment.
+
+> The same configuration variables can be used regardless of installation type
+> (container or package-based).
+
+### Variables
+
+#### SL_APP_HOST
+
+The `SL_APP_HOST` is the public-facing URL for the stoplight application.
```
Comment (rowa97) [inline articles/enterprise/components/app.md]: Caps 'Stoplight'

Comment (rossmcdonald) [reply articles/enterprise/components/app.md]: Fixed

File: articles/enterprise/components/gitlab.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+* Interfacing with the Stoplight API
+* User-facing notifications (including password reset emails, group/project
+  invitations, etc)
+* Tracking all changes between different files, and storing them within a Git
+  repository
+
+Packaged within the GitLab CE is an installation of PostgreSQL and Redis. These
+two sub-components can be broken out into external services if your organization
+is already familiar with running these (or similar) services. You may also break
+out these services if you plan on using a managed hosting solution, for example
```
Comment (rowa97) [inline articles/enterprise/components/gitlab.md]: for example,

File: articles/enterprise/components/gitlab.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+Amazon RDS (for PostgreSQL) or Amazon ElastiCache (for Redis).
+
+> ### Requirements
+>
+> #### Storage
+>
+> GitLab requires persistent storage in order to store Stoplight file data (in git), and optionally PostgreSQL data (when using the omnibus package).
```
Comment (rowa97) [inline articles/enterprise/components/gitlab.md]: cap Git

File: articles/enterprise/components/gitlab.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+>
+> #### Networking
+>
+> The default port for GitLab's HTTP API is TCP port 8080. The port can be customized via the GitLab configuration file. The GitLab process must also be able to initiate communication to PostgreSQL and Redis.
+>
+> _If not using the omnibus package, make sure that Redis and PostgreSQL are up and available prior to starting GitLab_
```
Comment (rowa97) [inline articles/enterprise/components/gitlab.md]: awk phrased "If not using"

File: articles/enterprise/components/gitlab.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+
+```bash
+sudo gitlab-ctl status
+```
+
+### Docker Installations
+
+#### Configuring the Container
+
+The GitLab container should be configured nearly identically to the package installation described above, via file. The easiest way to do this is to mount the GitLab configuration directory inside the container.
```
Comment (rowa97) [inline articles/enterprise/components/gitlab.md]: via file?

Comment (rossmcdonald) [reply articles/enterprise/components/gitlab.md]: Removed.

File: articles/enterprise/components/hub-builder.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+name=Stoplight Package Repository
+baseurl=https://$REPO_USERNAME:$REPO_PASSWORD@pkg.stoplight.io/rpm
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://pkg.stoplight.io/stoplight.key
+EOF
+```
+
+> Be sure to set your repository credentials before issuing the `cat` command
```
Comment (rowa97) [inline articles/enterprise/components/hub-builder.md]: I am very intrigued for some reason, of this 'cat' command

File: articles/enterprise/components/hub-builder.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+#### Installing the Hub Builder Package
+
+Once the repository is configured properly, you can install the Hub Builder
+component using the command:
+
+```bash
+sudo yum install stoplight-hub-builder -y
```
Comment (rowa97) [inline articles/enterprise/components/hub-builder.md]: and sudo yum as well, just partially yum

File: articles/enterprise/components/prism.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+name=Stoplight Package Repository
+baseurl=https://$REPO_USERNAME:$REPO_PASSWORD@pkg.stoplight.io/rpm
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://pkg.stoplight.io/stoplight.key
+EOF
+```
+
+> Be sure to set your repository credentials before issuing the `cat` command
```
Comment (rowa97) [inline articles/enterprise/components/prism.md]: there it is again, slinking about

File: articles/enterprise/components/hub-builder.md (hunk 13 above)
Comment (rossmcdonald) [reply articles/enterprise/components/hub-builder.md]: Hah, it took me too long to get that.

File: articles/enterprise/components/prism.md [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+If started properly, the container should be marked with a healthy status after
+30 seconds. Use the `docker ps` command to verify the container was started and
+is running properly.
+
+## Post-install Validations
+
+Once the Prism component is running, you can verify the installation was
+successful issuing an HTTP GET request to the `/health` URL on the HTTP admin
```
[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/prism.md]: Fixed

File: articles/enterprise/components/pubs.md [hunk 16]
```diff
@@ -0,0 +1,269 @@
+# Pubs (Hubs Server)
+
+The **Pubs** component serves and catalogs hubs published through Stoplight.
```
[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/pubs.md]: Fixed

File: articles/enterprise/components/hub-builder.md (hunk 12 above)
Comment (rossmcdonald) [reply articles/enterprise/components/hub-builder.md]: I updated these block-quotes slightly to reference that the 🐱 command is the command above.

File: articles/enterprise/components/pubs.md [hunk 17]
```diff
@@ -0,0 +1,269 @@
...
+The Pubs default configuration is located at the path:
+
+```bash
+/etc/pubs/pubs.yml
+```
+
+Be sure to customize any variables as needed to match your environment before
+starting the Pubs service.
+
+> Any changes to the Pubs configuration require a service restart in order to
```
Comment (rowa97) [inline articles/enterprise/components/pubs.md]: requires

File: articles/enterprise/components/pubs.md [hunk 18]
```diff
@@ -0,0 +1,269 @@
...
+
+### RPM Package
+
+The Pubs default configuration is located at the path:
+
+```bash
+/etc/pubs/pubs.yml
+```
+
+Be sure to customize any variables as needed to match your environment before
```
Comment (rowa97) [inline articles/enterprise/components/pubs.md]: bold before

File: articles/enterprise/components/tasker.md [hunk 19]
```diff
@@ -0,0 +1,230 @@
+# Tasker (Jobs Server)
+
+The **Tasker** component runs scheduled and on-demand tasks for the Stoplight platform.
+
+> #### Networking Details
+>
+> The default port for the Tasker component is TCP port **9432**. This port can
+> be customzied via configuration variable.
```
Comment (rowa97) [inline articles/enterprise/components/tasker.md]: 'variables' if not proper noun

File: articles/enterprise/components/tasker.md [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+
+```
+TASKER_MODE="shell"
+```
+
+> If not specified, Tasker defaults to using `docker` mode.
+
+#### CORE_ROOT
+
+The `CORE_ROOT` denotes the absolutel path to stoplight-hub-builder package root.
```
Comment (rowa97) [inline articles/enterprise/components/tasker.md]: absolute

Comment (rowa97) [reply articles/enterprise/components/tasker.md]: 'the' stoplight-hub-builder package root

File: articles/enterprise/components/pubs.md (hunk 17 above)
Comment (rossmcdonald) [reply articles/enterprise/components/pubs.md]: Fixed

File: articles/enterprise/components/pubs.md (hunk 18 above)
Comment (rossmcdonald) [reply articles/enterprise/components/pubs.md]: Fixed

File: articles/enterprise/components/tasker.md (hunk 19 above)
Comment (rossmcdonald) [reply articles/enterprise/components/tasker.md]: Fixed

File: articles/enterprise/components/tasker.md (hunk 20 above)
Comment (rossmcdonald) [reply articles/enterprise/components/tasker.md]: Fixed

File: articles/enterprise/components/tasker.md [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+
+Be sure to customize any variables as needed to match your environment before
+starting the Tasker service.
+
+> Any changes to the Tasker configuration require a service restart in order to
+> take effect.
+
+### Docker
+
+To expose these to the Docker runtime, either write them to a file and use the
```
[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/tasker.md]: Fixed

File: articles/enterprise/components/tasker.md [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+Once the Tasker component is running, you can verify the installation was
+successful issuing an `HTTP GET` request to the `/health` endpoint:
+
+```bash
+# remember to update the scheme, host, and port here to match your installation
+curl -v http://localhost:9432/health
+```
+
+If Tasker was installed and configured properly, you will receive an `HTTP 200`
+reponse back.
```
[the remark this answers is not in the document]
Comment (rossmcdonald) [reply articles/enterprise/components/tasker.md]: Fixed

File: articles/enterprise/overview.md [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+Due to these shortcomings, single-server deployments are only recommended for
+POC, pilot, or trial environments.
+
+### Multi-Server
+
+Multi-server deployments run different Stoplight Enterprise components on
+separate Linux instances. This deployment option is much more resilient to
+system-level issues, though it does require more network configuration.
+
+Stoplight recommends multi-server deployments for all production-facing
```
Comment (rowa97) [inline articles/enterprise/overview.md]: production-facing? not just production?

Comment (rossmcdonald) [reply articles/enterprise/overview.md]: Yeah, it doesn't make sense. Fixed.

File: articles/enterprise/overview.md [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+* [InfluxDB](https://www.influxdata.com/time-series-platform/influxdb/) v1.3
+  for metrics storage and aggregation
+* [Kapacitor](https://www.influxdata.com/time-series-platform/kapacitor/) v1.3
+  for alerting and metrics processing
+* [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/) v1.4
+  for metrics collection
+* [Mtail](https://github.com/google/mtail) v3.0 for whitebox monitoring of
+  application logs
+
+Please note that the above recommendations are entirely optional if your
```
Comment (rowa97) [inline articles/enterprise/overview.md]: callout

**Pull request merged**